docs/cross-compilation.md

```diff
@@ -113,43 +131,49 @@ scala_library(
 See complete documentation in the [scala_cross_version_select.bzl](/scala/scala_cross_version_select.bzl) file
 
 #### Manually
+
 An example usage of `select()` to provide custom dependency for specific Scala version:
-```starlark
+
+```py
 deps = select({
     "@io_bazel_rules_scala_config//:scala_version_3_3_1": [...],
     ...
 })
 ```
 
 For more complex logic, you can extract it to a `.bzl` file:
-```starlark
+
+```py
 def srcs(scala_version):
     if scala_version.startswith("2"):
         ...
     ...
 ```
+
 and then in the `BUILD` file:
-```starlark
-load("@io_bazel_rules_scala//:scala_cross_version.bzl", "version_suffix")
-load("@io_bazel_rules_scala_config//:config.bzl", "SCALA_VERSIONS")
+
+```py
 load("....bzl", "srcs")
+load("@io_bazel_rules_scala_config//:config.bzl", "SCALA_VERSIONS")
+load("@rules_scala//:scala_cross_version.bzl", "version_suffix")
 
 scala_library(
     ...
     srcs = select({
         "@io_bazel_rules_scala_config//:scala_version" + version_suffix(v): srcs(v)
         for v in SCALA_VERSIONS
-    }), 
+    }),
     ...
 )
 ```
 
-
 ## Requesting specific version
+
 To use other than default version of Scala, you need to change the current `@io_bazel_rules_scala_config//:scala_version` build setting.
```

Comment:
Probably this line also needs to change.

Reply:
Pulling this thread destroyed my sweater... I've overhauled the entire document in https://github.com/bazelbuild/rules_scala/pull/1703/commits/427e1c9f92e4ae8252214565c5542e1a21a849e9. Please take a look and tell me what you think. I think most of it is OK, but I didn't do much to the "Cross-build support tiers" section at the end.

In the process, I learned that ['incompatible_use_toolchain_transition' has been deprecated since before Bazel 6.5.0](https://bazel.build/versions/6.4.0/rules/lib/globals?hl=en#rule). I took the liberty to remove that attribute throughout the project, and made sure `./test_all.sh` passed before including it in the same commit.

Reply:
I looks great, thanks!